### ISSUE TITLE
Make sure we generate the correct helm chart

### ISSUE THREAD

**Issue by mudler** [rancher/elemental-operator#13]
## Problem

We have upstream operator images and OBS images, but we want to generate one chart only that can be used for both ( the image should be tweakable from the values file).

## Action item

- Make sure we build from OBS the chart pointing to the release tagged images
- Make sure charts from github releases workflows are working as well
- Have a "-dev" chart pointing to master
- Setup CI workflow for releases (GH/OBS)
- Make sure all images referenced in operator are actually existing after merge of #3
- help @kkaempf with OBS helm chart packaging
- publish helm chart on `registry.suse.com`

**Comment by Itxaka** [comment]
helm chart allows for overriding of the repo and the tag on chart build by setting: `REPO=ttl.sh/operator TAG=latest` before running `make chart`

Currently on CI for PR/Master this is set to `REPO=quay.io/costoolkit/elemental-operator-ci` and `TAG=${GIT_TAG}-${GIT_COMMIT_SHORT}` for ease of use.

For release is set to `REPO=quay.io/costoolkit/elemental-operator` and `TAG=${GITHUB_REF##*/}` to be in sync with the images published.

So in this case, I would keep the master and PRs as is, would not change master to point always to `-dev` as that makes the pipeline more complex when its not really needed.

I would also keep releases as is, as it seemed to work as expected (we would need to push a new release to check that we are not missing anything.

In any case, IIRC those values are overridable on chart install by setting `--set "image.repository=REPO"` and `--set "image.tag=TAG"` so the end user can point to whatever image they want on any version of the chart.

So I would say that points 2,3,4 are done missing a new release to check that everything goes as expected.

Point 5 was done today when the operator was updated and tests fixed to the new code, all images worked as expected.

Basically missing the OBS stuff, but that should be a simple workflow to trigger the building process on OBS using the new REPO/TAG to build the chart AND the images.

**Comment by kkaempf** [comment]
Actually, the helm char of `rancher-operator` allowed overriding of the repo. I was not successful in overriding the repo for the `elemental-operator` chart. Was this fixed lately ?
The chart build is at line 73ff of [elemental-operator.spec](https://build.opensuse.org/package/view_file/isv:Rancher:Elemental/elemental-operator/elemental-operator.spec?expand=1)

**Comment by Itxaka** [comment]
@kkaempf yes, somehow it was removed while merging the systemd so no matter what REPO/TAG you set it would not pick it up. That was fixed on PR https://github.com/rancher/elemental-operator/pull/18/ merged today so update to latest version and retry adn you should be able to build the chart correctly with the default params that you want!

**Comment by Itxaka** [comment]
FYI, we can introduce a check for this in CI that basically builds the chart with some X values for those and then we can check with `helm show values` and check those values ended in the built chart

**Comment by mjura** [comment]
I have build initial helm chart for `elemental-operator` on OBS https://build.opensuse.org/package/show/home:mjura:branches:isv:Rancher:Elemental/elemental-operator-helm

**Comment by mjura** [comment]
Adding README to elemtal-operator helm chart https://github.com/rancher/elemental-operator/pull/56
